src/types/trace_filtering.rs

```diff
@@ -0,0 +1,465 @@
+    /// From block
+    #[serde(rename="fromBlock", skip_serializing_if = "Option::is_none")]
+    from_block: Option<BlockNumber>,
+    /// To block
+    #[serde(rename="toBlock", skip_serializing_if = "Option::is_none")]
+    to_block: Option<BlockNumber>,
+    /// From address
+    #[serde(rename="fromAddress", skip_serializing_if = "Option::is_none")]
+    from_address: Option<Vec<Address>>,
+    /// To address
+    #[serde(rename="toAddress", skip_serializing_if = "Option::is_none")]
+    to_address: Option<Vec<Address>>,
+    /// Output offset
+    #[serde(skip_serializing_if = "Option::is_none")]
+    after: Option<usize>,
+    /// Output amount
+    #[serde(skip_serializing_if = "Option::is_none")]
+    count: Option<usize>,
+}
+
+/// Trace Filter Builder
+#[derive(Default, Debug, Clone, PartialEq)]
+pub struct TraceFilterBuilder {
+    filter: TraceFilter,
+}
+
+impl TraceFilterBuilder {
+
+    /// Sets From block
+    pub fn from_block(mut self, block: BlockNumber) -> Self {
+        self.filter.from_block = Some(block);
+        self
+    }
+
+    /// Sets to block
+    pub fn to_block(mut self, block: BlockNumber) -> Self {
+        self.filter.to_block = Some(block);
+        self
+    }
+
+    /// Sets to address
+    pub fn to_address(mut self, address: Vec<H160>) -> Self {
+        self.filter.to_address = Some(address);
+        self
+    }
+
+    /// Sets from address
+    pub fn from_address(mut self, address: Vec<H160>) -> Self {
+        self.filter.from_address = Some(address);
+        self
+    }
+
+    /// Sets after offset
+    pub fn after(mut self, after: usize) -> Self {
+        self.filter.after = Some(after);
+        self
+    }
+
+    /// Sets amount of traces to display
+    pub fn count(mut self, count: usize) -> Self {
+        self.filter.count = Some(count);
+        self
+    }
+
+    /// Builds the Filter
+    pub fn build(&self) -> TraceFilter {
+        self.filter.clone()
+    }
+}
+
+// `LocalizedTrace` in Parity
+/// Trace-Filtering API trace type
+#[derive(Debug, PartialEq, Clone, Serialize)]
+pub struct Trace {
+    /// Action
+    pub action: Action,
+    /// Result
+    pub result: Res,
+    /// Trace address
+    pub trace_address: Vec<usize>,
+    /// Subtraces
+    pub subtraces: usize,
+    /// Transaction position
+    pub transaction_position: Option<usize>,
+    /// Transaction hash
+    pub transaction_hash: Option<H256>,
+    /// Block Number
+    pub block_number: u64,
+    /// Block Hash
+    pub block_hash: H256,
+}
+
+macro_rules! de_value {
+    ($action: ident) => ({
+        serde_json::from_value($action).map_err(|e| de::Error::custom(e.to_string()))
+    })
+}
+// a pretty standard custom deserialize, except it deserializes 'error' and 'result' of JSON
+// into the result enum, as well as deserializes `Action` based upon `type` field of the JSON.
+impl<'de> Deserialize<'de> for Trace {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error> where D: Deserializer<'de> {
```

Comment:
Please add some more tests to this deserializer, we should handle possible error cases as well, cause the stringy code seems quite fragile.